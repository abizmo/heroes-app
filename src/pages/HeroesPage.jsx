import React from 'react'

const HeroesPage = () => {
  return (
    <div>
      HeroesPage
    </div>
  )
}

export default HeroesPage
