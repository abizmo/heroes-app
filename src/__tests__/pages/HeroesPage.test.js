import { mount } from "enzyme"
import { Route, Router } from "react-router"
import {createMemoryHistory} from 'history'

import HeroesPage from "../../pages/HeroesPage"

describe('HeroesPage tests', () => {
  let history
  beforeEach(() => {
    history = createMemoryHistory({ initialEntries: ["/heroes/marvel-wolverine"]})
  })

  test('should redirect to /marvel if no hero', () => {
    const wrapper = mount(
      <Router history={history}>
        <HeroesPage />
      </Router>
    )
    
    expect(wrapper.find('Redirect').exists()).toBeTruthy()
  })
  
  test('should render and match snapshot', () => {
    const wrapper = mount(
      <Router history={history}>
        <Route path={"/heroes/:heroId"}>
          <HeroesPage />
        </Route>
      </Router>
    )
    
    expect(wrapper.find('h2').text()).toBe("Wolverine")
    expect(wrapper).toMatchSnapshot()
  })
  
  test('should goBack when click back', () => {
    history.push = jest.fn()
    const wrapper = mount(
      <Router history={history}>
        <Route path={"/heroes/:heroId"}>
          <HeroesPage />
        </Route>
      </Router>
    )
    
    wrapper.find('button').simulate('click')
    expect(history.push).toHaveBeenCalledWith('/')
  })
  
  test('should goto /marvel if history.length === 1 when click back', () => {
    const history = createMemoryHistory({ initialEntries: ["/heroes/marvel-wolverine","/marvel"]})
    history.goBack = jest.fn()
    const wrapper = mount(
      <Router history={history}>
        <Route path={"/heroes/:heroId"}>
          <HeroesPage />
        </Route>
      </Router>
    )
    
    wrapper.find('button').simulate('click')
    expect(history.goBack).toHaveBeenCalled()
  })
})
