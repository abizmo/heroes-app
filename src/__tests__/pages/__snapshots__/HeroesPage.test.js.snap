// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesPage tests should render and match snapshot 1`] = `
<div
  className="container py-5"
>
  <div
    className="row animate__animated animate__fadeIn"
  >
    <div
      className="col-md-5"
    >
      <img
        alt="Wolverine"
        className="img-fluid rounded-start"
        src="../assets/heroes/marvel-wolverine.jpg"
      />
    </div>
    <div
      className="col-md-7"
    >
      <h2
        className="featurette-heading mb-5"
      >
        Wolverine
      </h2>
      <ul
        className="list-group list-group-flush mb-4"
      >
        <li
          className="list-group-item"
        >
          <b>
            Alter ego: 
          </b>
          James Howlett
        </li>
        <li
          className="list-group-item"
        >
          <b>
            Publisher: 
          </b>
          Marvel Comics
        </li>
        <li
          className="list-group-item"
        >
          <b>
            First appearance: 
          </b>
          The Incredible Hulk #180
        </li>
      </ul>
      <h5
        className="featurette-heading text-muted"
      >
        Characters
      </h5>
      <p
        className="lead mb-5"
      >
        James Howlett
      </p>
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
